src/App/Fossa/Config/Report.hs

```diff
@@ -40,20 +43,45 @@ import Options.Applicative (
   option,
   optional,
   progDesc,
+  strOption,
   switch,
  )
 import Path.IO (getCurrentDir)
+import Text.Read (readMaybe)
 
 data ReportType = Attribution deriving (Eq, Ord, Enum, Bounded)
 
 instance Show ReportType where
   show Attribution = "attribution"
 
--- TODO: Add support for text-format reports
 data ReportOutputFormat
   = ReportJson
-  -- ReportPretty
-  deriving (Eq, Ord, Show)
+  | ReportMarkdown
+  | ReportSpdx
+  deriving (Eq, Ord, Enum, Bounded)
+
+instance Read ReportOutputFormat where
+  readsPrec _ s | s == show ReportJson = [(ReportJson, safeStripPrefix (show ReportJson) s)]
+  readsPrec _ s | s == show ReportSpdx = [(ReportSpdx, safeStripPrefix (show ReportSpdx) s)]
+  readsPrec _ s | s == show ReportMarkdown = [(ReportMarkdown, safeStripPrefix (show ReportMarkdown) s)]
+  readsPrec _ _ = []
+
+safeStripPrefix :: String -> String -> String
+safeStripPrefix a b = M.fromMaybe "" $ stripPrefix a b
+
+instance ToText ReportOutputFormat where
+  toText = toText . show
+
+instance Show ReportOutputFormat where
+  show ReportJson = "json"
+  show ReportMarkdown = "markdown"
+  show ReportSpdx = "spdx"
+
+reportOutputFormatList :: String
+reportOutputFormatList = intercalate ", " $ map show allFormats
+  where
+    allFormats :: [ReportOutputFormat]
```

Comment:
This is `Data.List.Extra.enumerate`.

Reply:
That function doesn't exist.

Reply:
https://github.com/fossas/fossa-cli/blob/master/src/Data/List/Extra.hs

Reply:
I assume this is from a custom prelude package.  We've had trouble using those, it can kill IDE inference, and has made it harder for newcomers to get involved in the codebase.

Reply:
Though it's been a while since we've tried using one, and the one we used was hand-rolled.